packages/react-native-reanimated/plugin/src/workletFactory.ts

```diff
@@ -325,17 +325,44 @@ function hash(str: string): number {
   return (hash1 >>> 0) * 4096 + (hash2 >>> 0);
 }
 
-function makeWorkletName(fun: NodePath<WorkletizableFunction>): string {
+let functionId = 1;
+
+function makeWorkletName(
+  fun: NodePath<WorkletizableFunction>,
+  state: ReanimatedPluginPass
+): string {
+  let source = 'unknown_file';
+
+  if (state.file.opts.filename) {
+    const filepath = state.file.opts.filename;
+    // Take the file name without extension.
+    source = basename(filepath).split('.')[0];
+    const splitFilepath = filepath.split('/');
+    // Get the library name from the path.
+    const nodeModulesIndex = splitFilepath.indexOf('node_modules');
+    if (nodeModulesIndex !== -1) {
+      // Remove all non-alphanumeric characters.
+      const libraryName = splitFilepath[nodeModulesIndex + 1].replace(
+        /\W/g,
+        ''
+      );
+      source = libraryName + '_' + source;
+    }
+  }
+
+  const suffix = source + functionId++;
```

Comment:
Maybe just to be safe and prevent potentially strange errors when `source` is `undefined` or `null` (theoreticly `source = basename(filepath).split('.')[0];` can produce `undefined` for a file without extension), let's replace it with the following syntax:
```js
const suffix = `${source}${functionId++}`;
```

But if you think that is overkill you don't need to 😅

Reply:
@piaskowyk `source` is initialized with `unknown_file` in line 334 😅

Reply:
Also nothing bad will happen if there's no extension.

<img width="267" alt="Screenshot 2024-07-03 at 09 26 17" src="https://github.com/software-mansion/react-native-reanimated/assets/40713406/8de75aff-f5a5-4224-9769-a2546e96fc4b">

Reply:
ok, you convinced me 😅